corehq/apps/data_cleaning/views/status.py

```diff
@@ -39,6 +39,22 @@ def seconds_since_complete(self):
     def is_session_in_progress(self):
         return self.session.committed_on is not None and self.seconds_since_complete < APPLY_CHANGES_WAIT_TIME
 
+    @property
+    def weighted_percent_complete(self):
+        """
+        This gives the user the illusion of progress while we artificially buffer the completion time.
+
+        The buffer allows the change feed to catch up to the form submissions,
+        so that the user doesn't refresh and see that their data hasn't updated due to a slow change feed.
+
+        TODO: update this buffer (APPLY_CHANGES_WAIT_TIME) dynamically based on change feed status.
```

Comment:
@AmitPhulera did some work on a PillowLagGaugeLimiter fyi. If nothing else would serve as a useful reference for pulling lag time.